pokemon/serializers.py

```
@@ -0,0 +1,115 @@
+from typing import Any
+
+import requests
+from rest_framework import serializers
+
+from .models import Pokemon, PokemonAbility, PokemonType
+
+
+class PokemonSerializer(serializers.ModelSerializer):
+    pokemon_id = serializers.IntegerField(required=False)
+    name = serializers.CharField(required=False)
+    is_default = serializers.CharField(required=False)
+    base_experience = serializers.CharField(required=False)
+    height = serializers.CharField(required=False)
+    weight = serializers.CharField(required=False)
+    abilities_list = serializers.SerializerMethodField(required=False, read_only=True)
+    types_list = serializers.SerializerMethodField(required=False, read_only=True)
+
+    def get_abilities_list(self, pokemon: Pokemon):
+        return list(pokemon.abilities.all().values())
+
+    def get_types_list(self, pokemon: Pokemon):
+        return list(pokemon.types.all().values())
+
+    class Meta:
+        model = Pokemon
+        fields = (
+            'pokemon_id',
+            'name',
+            'is_default',
+            'base_experience',
+            'height',
+            'weight',
+            'types_list',
+            'abilities_list',
+        )
+
+    def create(self, data: 'dict[str, Any]') -> Pokemon:
+        # TODO: instead of pulling all at once
+        # pull them in batches
+        url = 'https://pokeapi.co/api/v2/pokemon/?offset=0&limit=1200'
+
+        try:
+            all_pokemons = requests.get(url).json()['results']
+        except Exception as e:  # noqa
+            print('Raise error')
```

maybe should just raise the error instead of simply printing something